aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -0,0 +1,87 @@
+#include <ATen/BatchedTensorImpl.h>
+#include <torch/library.h>
+#include <ATen/BatchingUtils.h>
+#include <ATen/ATen.h>
+
+namespace at {
+
+// Note: [How to write batching rules]
+//
+// Whenever you do a vmap, on a vmap'ed tensor, the dimension that is being vmap'ed
```

Comment:
This paragraph sounds a bit strange. Maybe this is better (not a native speaker):

"""For each level of vmap the specified dimension is recorded. Once the innermost vmap was reached the recorded batch dimensions act as broadcasting dimensions for the given operation."""